pulsar/src/main/java/io/micronaut/pulsar/annotation/PulsarConsumer.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.pulsar.annotation;
+
+import io.micronaut.messaging.annotation.MessageListener;
+import io.micronaut.pulsar.config.AbstractPulsarConfiguration;
+import org.apache.pulsar.client.api.RegexSubscriptionMode;
+
+import javax.validation.constraints.Pattern;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Marks a method as a Pulsar Consumer.
+ *
+ * @author Haris Secic
+ * @since 1.0
+ */
+@Documented
+@Retention(RUNTIME)
+@Target({ElementType.METHOD})
+@MessageListener
+public @interface PulsarConsumer {
+
+    /**
+     * @return List of topic names in form of (persistent|non-persistent)://tenant-name/namespace/topic
+     */
+    String[] topics() default {};
```

Comment:
Correct, I planned ahead but I really forgot because of other issues like trying to make the build work between Linux based server and my local machine being Windows. Spotless was main pain that cause \n problems so I was required to build on WSL. Should I drop `TOPIC_VALIDATOR` or can I leave it as unused for now? Or do you want to wait until I make the annotation to use it?